src/main/java/uk/gov/hmcts/reform/managecase/service/NotifyService.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.hmcts.reform.managecase.service;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import java.net.ConnectException;
+import java.util.List;
+import java.util.Map;
+import javax.validation.ValidationException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.managecase.ApplicationParams;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientException;
+import uk.gov.service.notify.SendEmailResponse;
+
+@Service
+public class NotifyService {
+
+    private final NotificationClient notificationClient;
+    private final ApplicationParams appParams;
+
+    @Autowired
+    public NotifyService(ApplicationParams appParams,
+                         NotificationClient notificationClient) {
+        this.notificationClient = notificationClient;
+        this.appParams = appParams;
+    }
+
+    public List<SendEmailResponse> senEmail(final List<String> caseIds,
+                                            final List<String> emailAddresses) throws NotificationClientException  {
+        if (caseIds == null || caseIds.isEmpty()) {
+            throw new ValidationException("At least one case id is required to send notification");
+        }
+
+        if (emailAddresses == null || emailAddresses.isEmpty()) {
+            throw new ValidationException("At least one email address is required to send notification");
+        }
+
+        List<SendEmailResponse> emailNotificationResponses = Lists.newArrayList();
+        for (String caseId : caseIds) {
+            for (String emailAddress : emailAddresses) {
+                emailNotificationResponses.add(sendNotification(caseId, emailAddress));
```

Comment:
Assume 10 notifications to be sent out.
7 success, 1 fail-after-3-retries, what will happen?

Reply:
remaining notifications going to be ignored

Reply:
Well, the method will throw an Exception but many notifications will have already been sent.
If the calling contexts retry the whole batch, it will re-send many notifications. Please give some further thinking and implementation to error handling. We can discuss on a quick call.